trinity/extensibility/plugin_manager.py

```diff
@@ -63,7 +96,7 @@ def broadcast(self, event: BaseEvent, exclude: BasePlugin = None) -> None:
         """
         for plugin in self._plugin_store:
 
-            if plugin is exclude:
+            if plugin is exclude or not self._is_responsible_for_plugin(plugin):
```

Comment:
This `exclude` API stands out as something that probably needs an iteration to cleanup.

- it seems to only be used by the `PluginManager` itself, indicating that the API shouldn't be public.
- it only supports exclusion of a single plugin which seems arbitrarily inflexible.

Not for this PR necessarily but probably something worth cleaning up.

Reply:
Agreed. That whole API is entirely in flux. I'll get back to that as soon as the dust settles.